src/Plug.vala

```diff
@@ -47,6 +47,20 @@ namespace Power {
                 main_view = new BehaviorView ();
                 stack.add_named (main_view, "Power");
 
+                if (main_view.battery.is_present ()) {
+                  var icon_image = new Gtk.Image.from_icon_name ("computer-laptop", Gtk.IconSize.DIALOG);
```

Comment:
I know you're probably just prototyping/iterating, but check the About plug, I believe these lines get the device icon from DBus (e.g. a desktop icon on a desktop computer, a laptop icon on a laptop): https://github.com/elementary/switchboard-plug-about/blob/0e68edabe6210aa2685a72233e11d5fda4aa2efe/src/Views/HardwareView.vala#L164-L176

Reply:
Implemented! Thanks @cassidyjames